homeassistant/components/homekit/type_security_systems.py

```diff
@@ -56,13 +70,70 @@ def __init__(self, *args):
         state = self.hass.states.get(self.entity_id)
         self._alarm_code = self.config.get(ATTR_CODE)
 
+        supported_states = state.attributes.get(
+            "supported_features",
+            (
+                SUPPORT_ALARM_ARM_HOME
+                | SUPPORT_ALARM_ARM_AWAY
+                | SUPPORT_ALARM_ARM_NIGHT
+                | SUPPORT_ALARM_TRIGGER
+            ),
+        )
+
+        loader = get_loader()
+        default_current_states = loader.get_char(
+            "SecuritySystemCurrentState"
+        ).properties.get("ValidValues")
+        default_target_services = loader.get_char(
+            "SecuritySystemTargetState"
+        ).properties.get("ValidValues")
+
+        current_supported_states = [
+            HASS_TO_HOMEKIT[STATE_ALARM_DISARMED],
+            HASS_TO_HOMEKIT[STATE_ALARM_TRIGGERED],
+        ]
+        target_supported_services = [HASS_TO_HOMEKIT_SERVICES[SERVICE_ALARM_DISARM]]
+
+        if supported_states & SUPPORT_ALARM_ARM_HOME:
+            current_supported_states.append(HASS_TO_HOMEKIT[STATE_ALARM_ARMED_HOME])
+            target_supported_services.append(
+                HASS_TO_HOMEKIT_SERVICES[SERVICE_ALARM_ARM_HOME]
+            )
+
+        if supported_states & SUPPORT_ALARM_ARM_AWAY:
+            current_supported_states.append(HASS_TO_HOMEKIT[STATE_ALARM_ARMED_AWAY])
+            target_supported_services.append(
+                HASS_TO_HOMEKIT_SERVICES[SERVICE_ALARM_ARM_AWAY]
+            )
+
+        if supported_states & SUPPORT_ALARM_ARM_NIGHT:
+            current_supported_states.append(HASS_TO_HOMEKIT[STATE_ALARM_ARMED_NIGHT])
+            target_supported_services.append(
+                HASS_TO_HOMEKIT_SERVICES[SERVICE_ALARM_ARM_NIGHT]
+            )
+
+        new_current_states = {
+            key: val
+            for key, val in default_current_states.items()
+            if val in current_supported_states
+        }
+        new_target_services = {
+            key: val
+            for key, val in default_target_services.items()
+            if val in target_supported_services
+        }
+
         serv_alarm = self.add_preload_service(SERV_SECURITY_SYSTEM)
         self.char_current_state = serv_alarm.configure_char(
-            CHAR_CURRENT_SECURITY_STATE, value=3
+            CHAR_CURRENT_SECURITY_STATE, value=3, valid_values=new_current_states
         )
         self.char_target_state = serv_alarm.configure_char(
-            CHAR_TARGET_SECURITY_STATE, value=3, setter_callback=self.set_security_state
+            CHAR_TARGET_SECURITY_STATE,
+            value=3,
+            valid_values=new_target_services,
+            setter_callback=self.set_security_state,
```

Comment:
I think this is going to throw an exception if `3` is not a valid value.

Reply:
Actually I don't think that can ever happen

Reply:
3 is disarmed, so yeah it can never happen!